launcher.py

```diff
@@ -0,0 +1,405 @@
+import argparse
+import atexit
+import json
+import os
+import platform
+import subprocess
+import sys
+import time
+import webbrowser
+from pathlib import Path
+
+CONFIG_FILE = os.path.join("config.json")
+venv_name = "virtual_env"  # For downloading python packages
+venv_dir = Path.cwd() / venv_name  # Generate the path for the virtual environment
+
+
+class DebugLevel:
+    """Enum for debug levels"""
+
+    NONE = 0
+    APP = 1
+    ALL = 2
+
+    _string_to_debug_level = {"none": NONE, "app": APP, "all": ALL}
+
+
+current_debug_level = DebugLevel.NONE
+
+
+def run_command(command, cwd=None):
+    """
+    Executes a shell command.
+
+    Args:
+        command (str): The command to run.
+
+    Raises:
+        subprocess.CalledProcessError: If any command fails.
+    """
+    if current_debug_level == DebugLevel.ALL:
+        print(f"Executing command: {command}")
+    subprocess.check_call(
+        command.split(),
+        cwd=cwd,
+        stdout=None if current_debug_level == DebugLevel.ALL else subprocess.DEVNULL,
+        stderr=None if current_debug_level == DebugLevel.ALL else subprocess.DEVNULL,
+    )
```

Comment:
Would be good to try except here and other places where it's quite possible that we get an exception.

Reply:
I purposefully do not handle exception here as exception is handled by the calling entity. This way we can customize exception handling.

Reply:
Yeah that makes sense, I missed you clearly document that in the doctoring as well. Let's keep it as is